lib/mix/lib/mix/tasks/test.ex

```diff
@@ -184,8 +184,12 @@ defmodule Mix.Tasks.Test do
     test_files   = parse_files(files, test_paths)
     test_pattern = project[:test_pattern] || "*_test.exs"
 
-    test_files = Mix.Utils.extract_files(test_files, test_pattern)
-    _ = Kernel.ParallelRequire.files(test_files)
+    case Mix.Utils.extract_files(test_files, test_pattern) do
+      [] ->
+        Mix.raise "No files available to compile"
```

Comment:
If we can include the files (not the test_pattern) in the error message, even better!